### PR TITLE
Daily forecast block

### PR DESCRIPTION
**Pull request by eric-gade**
## What does this PR do? 🛠️
Implements the Daily Forecast block with live API data, per Issue #178 

## What does the reviewer need to know? 🤔
To try it out locally:
- [ ] zap  the containers
- [ ] navigate to the [front page](http://localhost:8080)
- [ ] Enter a location of your choice in the location search and navigate
- [ ] Examine the Daily Forecast block below the hourly forecast block

## Sandbox preview link
(Note: if you are reading this a long time after the PR was filed, this link may no longer be relevant):
[Click here for a live preview in the sandbox](https://weathergov-eric.app.cloud.gov/local/OKX/34/34/New%20York)

## Screenshots (if appropriate): 📸
<details>
<summary>Mobile</summary>

Screen Shot 2023-11-28 at 2 27 20 PM

</details>

<details>
<summary>Desktop</summary>

Screen Shot 2023-11-28 at 2 27 07 PM

</details>

### CODE REVIEW

**Review by greg-does-weather — APPROVED** [review]
I think there's a bug in the timezones, where further western time zones might have their "what counts as tomorrow" answered by the New York time zone. I think the `$tomorrow` variable needs to be set to the forecast's timezone before the `->modify('tomorrow')` call. I've offered a suggested change to the text fixtures that will trigger the bug. That said, I'm okay with opening a bug issue and going ahead with merging this. It shouldn't be too bad to fix (there's some prior art in the hourly forecast).

Otherwise just some comments. Merge at your leisure!

## Code Review Checklist

A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [x] (if applicable) Post deploy steps are run
- [x] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [x] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [x] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [x] All form inputs have explicit labels
    - [x] Form instructions are associated with inputs
    - [x] All relevant images use an img tag
    - [x] All images have appropriate alt attributes
    - [x] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [x] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [x] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [x] Tab order is logical
    - [x] Tables are coded properly. Tables have proper headers and column attributes
    - [x] Headings are nested properly. Heading elements are nested in a logical way
    - [x] Language is set. The language for the page is set
    - [x] CSS is not required to use the page. The page makes sense with or without CSS
    - [x] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [x] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [x] chrome/chromium/edge (renders correctly and user interactions work)
- [x] safari/webkit (renders correctly and user interactions work)
- [x] web page is readable and usable 
    - [x] at 480px     (mobile)
    - [x] at 640px     (tablet)
    - [x] at 1024px    (desktop)

File: web/modules/weather_data/src/Service/WeatherDataService.php [hunk 1]
```diff
@@ -269,4 +346,86 @@ public function getHourlyForecast($route, $now = FALSE) {
...
+    // route URI.
+    $wfo = $route->getParameter("wfo");
+    $gridX = $route->getParameter("gridX");
+    $gridY = $route->getParameter("gridY");
+
+    $forecast = $this->client->get("https://api.weather.gov/gridpoints/$wfo/$gridX,$gridY/forecast");
+    $forecast = json_decode($forecast->getBody());
+
+    $periods = $forecast->properties->periods;
+    $periods = $this->filterToFutureDays($periods, $defaultDays, $now->modify('tomorrow'));
```
[the remark this answers is not in the document]
Comment (eric-gade) [reply web/modules/weather_data/src/Service/WeatherDataService.php]: You are totally right -- the trick is to set `$now` (from which "tomorrow" can be computed) to be the timestamp of the first reported period in a daily forecast.
  
I have changed that, along with the function signature (to be more logical) in https://github.com/weather-gov/weather.gov/pull/452/commits/00d80f20c9aeea2140fa5059d6ca8101d435f273

File: web/modules/weather_data/src/Service/WeatherDataService.filterToFutureDays.php.test [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    // Ensure that all periods returned are GTE tomorrow.
+    foreach ($filteredPeriods as $period) {
+      $startTime = \DateTimeImmutable::createFromFormat(
+        \DateTimeInterface::ISO8601_EXPANDED,
+        $period->startTime
+      );
+
+      // Note: the arguments here are the opposite of what
+      // you might think given the method name.
+      $this->assertGreaterThanOrEqual($tomorrow, $startTime);
```
[the remark this answers is not in the document]
Comment (eric-gade) [reply web/modules/weather_data/src/Service/WeatherDataService.filterToFutureDays.php.test]: I have altered the test so it's more logical, easier to read, and also separates the "limit" part to a second case: https://github.com/weather-gov/weather.gov/pull/452/commits/00d80f20c9aeea2140fa5059d6ca8101d435f273

File: web/modules/weather_data/src/Service/WeatherDataService.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (greg-does-weather) [reply web/modules/weather_data/src/Service/WeatherDataService.php]: Ooh, that's much better than what I was thinking. Now I'm thinking the hourly forecast component can get a similar update and remove an extra API call.

**Pull request merged**